File: src/main/java/net/usbwire/usbplus/USBPlus.java
package net.usbwire.usbplus;

import net.fabricmc.api.ClientModInitializer;
import net.fabricmc.loader.api.FabricLoader;
import net.minecraft.client.MinecraftClient;
import net.usbwire.usbplus.config.Config;
import net.usbwire.usbplus.commands.CommandUtil;
import net.usbwire.usbplus.util.MixinHelper;
import org.apache.logging.log4j.*;


public class USBPlus implements ClientModInitializer {
	// get this from gradle.properties
	public static final FabricLoader fabricLoader = FabricLoader.getInstance();
	public static final String modid =
			fabricLoader.getModContainer("usbplus").get().getMetadata().getId();
	public static final String name =
			fabricLoader.getModContainer("usbplus").get().getMetadata().getName();
	public static final Logger logger = LogManager.getLogger(modid);
	public static final String configPath = "./config/" + modid;
	public static final Config config = new Config();

	public static final MinecraftClient mc = MinecraftClient.getInstance();

	@Override
	public void onInitializeClient() {
		logger.info("Starting usbplus!");
		MixinHelper.init();
		CommandUtil.init();
		logger.info("Started usbplus!");
	}
}

File: src/main/kotlin/net/usbwire/usbplus/USBPlus.kt
package net.usbwire.usbplus

import net.fabricmc.api.ClientModInitializer
import net.fabricmc.loader.api.FabricLoader
import net.minecraft.client.MinecraftClient
import net.usbwire.usbplus.commands.ConfigCommand
import net.usbwire.usbplus.features.Poi
import net.usbwire.usbplus.util.MixinHelper
import org.slf4j.LoggerFactory

object USBPlus : ClientModInitializer {
  // get this from gradle.properties
  val fabricmodjson by lazy { FabricLoader.getInstance().getModContainer("usbplus").get() }
  val modid: String by lazy { fabricmodjson.metadata.id }
  val name: String by lazy { fabricmodjson.metadata.name }
  val configPath: String = "./config/${modid}"
  val logger = LoggerFactory.getLogger(modid)
  // https://github.com/Skytils/SkytilsMod/blob/268e8e473a00e55cddc89c47653c3a00db263aac/src/main/kotlin/gg/skytils/skytilsmod/Skytils.kt#L122
  @JvmStatic val mc: MinecraftClient by lazy { MinecraftClient.getInstance() }

  override fun onInitializeClient() {
    ConfigCommand.register()
    MixinHelper.init()
  }
}
